### ISSUE TITLE
Build failed on Flutter Master channel

### ISSUE THREAD

**Issue by guyluz11** [Baseflow/flutter_cached_network_image#870]
## 🐛 Bug Report

I have switched to the Flutter **`Master`** channel and I am getting the following error when trying to run `flutter build web`

Can't reproduce on Flutter `Stable` channel!

### Expected behavior
Compile my site normally

### Error
<details>

```dart

Target dart2js failed: ProcessException: Process exited abnormally:
../../../.pub-cache/hosted/pub.dev/cached_network_image-3.2.3/lib/src/image_provider/cached_network_image_provider.dart:78:54:
Error: Type 'DecoderCallback' not found.
      image_provider.CachedNetworkImageProvider key, DecoderCallback decode) {
                                                     ^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/cached_network_image-3.2.3/lib/src/image_provider/cached_network_image_provider.dart:99:5:
Error: Type 'DecoderCallback' not found.
    DecoderCallback decode,
    ^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/cached_network_image_platform_interface-2.0.0/lib/cached_network_image_platform_interface.dart:29:5:
Error: Type 'DecoderCallback' not found.
    DecoderCallback decode,
    ^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/cached_network_image_web-1.0.2/lib/cached_network_image_web.dart:25:5:
Error: Type 'DecoderCallback' not found.
    DecoderCallback decode,
    ^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/flutter_blurhash-0.7.0/lib/src/blurhash_widget.dart:218:42:
Error: Type 'DecoderCallback' not found.
  ImageStreamCompleter load(UiImage key, DecoderCallback decode) => OneFrameImageStreamCompleter(_loadAsync(key));
                                         ^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/flutter_blurhash-0.7.0/lib/src/blurhash_image.dart:32:48:
Error: Type 'DecoderCallback' not found.
  ImageStreamCompleter load(BlurHashImage key, DecoderCallback decode) => OneFrameImageStreamCompleter(_loadAsync(key));
                                               ^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/cached_network_image-3.2.3/lib/src/image_provider/cached_network_image_provider.dart:78:54:
Error: 'DecoderCallback' isn't a type.
      image_provider.CachedNetworkImageProvider key, DecoderCallback decode) {
                                                     ^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/cached_network_image-3.2.3/lib/src/image_provider/cached_network_image_provider.dart:99:5:
Error: 'DecoderCallback' isn't a type.
    DecoderCallback decode,
    ^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/cached_network_image_platform_interface-2.0.0/lib/cached_network_image_platform_interface.dart:29:5:
Error: 'DecoderCallback' isn't a type.
    DecoderCallback decode,
    ^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/cached_network_image_web-1.0.2/lib/cached_network_image_web.dart:25:5:
Error: 'DecoderCallback' isn't a type.
    DecoderCallback decode,
    ^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/flutter_blurhash-0.7.0/lib/src/blurhash_widget.dart:218:42:
Error: 'DecoderCallback' isn't a type.
  ImageStreamCompleter load(UiImage key, DecoderCallback decode) => OneFrameImageStreamCompleter(_loadAsync(key));
                                         ^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/flutter_blurhash-0.7.0/lib/src/blurhash_image.dart:32:48:
Error: 'DecoderCallback' isn't a type.
  ImageStreamCompleter load(BlurHashImage key, DecoderCallback decode) => OneFrameImageStreamCompleter(_loadAsync(key));
                                               ^^^^^^^^^^^^^^^
Error: Compilation failed.
  Command: /home/guyluz/Programs/flutter/bin/cache/dart-sdk/bin/dart --disable-dart-dev /home/guyluz/Programs/flutter/bin/cache/dart-sdk/bin/snapshots/dart2js.dart.snapshot --platform-binaries=/home/guyluz/Programs/flutter/bin/cache/flutter_web_sdk/kernel
  --invoker=flutter_tool -Ddart.vm.product=true -DFLUTTER_WEB_AUTO_DETECT=true -DFLUTTER_WEB_CANVASKIT_URL=https://www.gstatic.com/flutter-canvaskit/cb58abd773268d62008120f270759cb9a3904f11/ --native-null-assertions --no-source-maps -o
  /home/guyluz/Documents/git/CBJ_Site/.dart_tool/flutter_build/95587cefb7b011342d5c260323988f72/app.dill --packages=.dart_tool/package_config.json --cfe-only /home/guyluz/Documents/git/CBJ_Site/.dart_tool/flutter_build/95587cefb7b011342d5c260323988f72/main.dart
#0      RunResult.throwException (package:flutter_tools/src/base/process.dart:125:5)
#1      _DefaultProcessUtils.run (package:flutter_tools/src/base/process.dart:278:19)
<asynchronous suspension>
#2      Dart2JSTarget.build (package:flutter_tools/src/build_system/targets/web.dart:189:5)
<asynchronous suspension>
#3      _BuildInstance._invokeInternal (package:flutter_tools/src/build_system/build_system.dart:853:9)
<asynchronous suspension>
#4      Future.wait.<anonymous closure> (dart:async/future.dart:523:21)
<asynchronous suspension>
#5      _BuildInstance.invokeTarget (package:flutter_tools/src/build_system/build_system.dart:791:32)
<asynchronous suspension>
#6      Future.wait.<anonymous closure> (dart:async/future.dart:523:21)
<asynchronous suspension>
#7      _BuildInstance.invokeTarget (package:flutter_tools/src/build_system/build_system.dart:791:32)
<asynchronous suspension>
#8      FlutterBuildSystem.build (package:flutter_tools/src/build_system/build_system.dart:620:16)
<asynchronous suspension>
#9      WebBuilder.buildWeb (package:flutter_tools/src/web/compile.dart:89:34)
<asynchronous suspension>
#10     BuildWebCommand.runCommand (package:flutter_tools/src/commands/build_web.dart:198:5)
<asynchronous suspension>
#11     FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1323:27)
<asynchronous suspension>
#12     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
<asynchronous suspension>
#13     CommandRunner.runCommand (package:args/command_runner.dart:212:13)
<asynchronous suspension>
#14     FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:348:9)
<asynchronous suspension>
#15     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
<asynchronous suspension>
#16     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:294:5)
<asynchronous suspension>
#17     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:113:9)
<asynchronous suspension>
#18     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
<asynchronous suspension>
#19     main (package:flutter_tools/executable.dart:90:3)
<asynchronous suspension>

Compiling lib/main.dart for the Web...                              7.7s
Exception: Failed to compile application for the Web.

```

</details>

### Reproduction steps
.1 Move to Flutter `Master`
2. Build using `flutter build web`

### Configuration

**Version:**   `cached_network_image: ^3.2.3`

**Platform:** 
- [x] :iphone: iOS
- [x] :robot: Android
- [x] : Web

flutter --version
```
Flutter 3.14.0-9.0.pre.90 • channel master • https://github.com/flutter/flutter.git
Framework • revision 901a392ae5 (52 minutes ago) • 2023-08-25 07:23:26 -0400
Engine • revision cb58abd773
Tools • Dart 3.2.0 (build 3.2.0-97.0.dev) • DevTools 2.26.2
```

**Comment by MominRaza** [comment]
[DecoderCallback](https://api.flutter.dev/flutter/painting/DecoderCallback.html) is deprecated since Flutter 2.13 and it is now removed in https://github.com/flutter/flutter/pull/132679

**Comment by selvam920** [comment]
how please check this issue

**Comment by gauravmehta13** [comment]
Use this:
https://github.com/gauravmehta13/flutter_cached_network_image

**Comment by gauravmehta13** [comment]
```
  cached_network_image:
    git:
      url: https://github.com/gauravmehta13/flutter_cached_network_image.git
      path: cached_network_image
```

**Comment by MominRaza** [comment]
> Use this:
> https://github.com/gauravmehta13/flutter_cached_network_image
> 

You should have forked the original repo so we can see what changes you made. And you could also raise a PR.

**Comment by gauravmehta13** [comment]
> > Use this:
> > [gauravmehta13/flutter_cached_network_image](https://github.com/gauravmehta13/flutter_cached_network_image)
> 
> You should have forked the original repo so we can see what changes you made. And you could also raise a PR.

If i could have i would have, there were some issues with that, If you want to use then you are welcome else you can create your own fork, Thanks

**Comment by gauravmehta13** [comment]
I used https://github.com/Baseflow/flutter_cached_network_image/pull/861 as a reference, So you can just fork this and remove the deprecated code.

**Comment by antonengelhardt** [comment]
@gauravmehta13 Thanks for the hint! I also can't reproduce this on `stable`, but on `master` our builds are failing.

**Comment by T-P-F** [comment]
Now that flutter stable is updated, this will be happening to anyone who runs `flutter upgrade`.

**Comment by MominRaza** [comment]
> Now that flutter stable is updated, this will be happening to anyone who runs `flutter upgrade`.

No this change still only in master, not even beta
But it will be eventually land in Stable, @guyluz11 please fix it in mean time

**Comment by hossam-96** [comment]
Same issue here

**Comment by clublounge** [comment]
Same issue here

**Comment by guyluz11** [comment]
Guy's please like (👍) the issue if it effects you.
It will keep the issue cleaner then writing it effects me.

Thanks

**Comment by amake** [comment]
This is now affecting the beta channel.

**Comment by renefloor** [comment]
Fix in 3.3.0

**Comment by Siddharth2212** [comment]
Thanks!
